### PR TITLE
replica: serialize created-by reference

### PR DESCRIPTION
**Pull request by simon3z**
/cc @lavalamp

### CODE REVIEW

**Comment by simon3z** [comment]
@lavalamp this is not fully ready to be merged yet but please check that the general idea is ok. Thanks!

File: pkg/api/types.go [hunk 1]
```diff
@@ -1583,6 +1583,11 @@ type ObjectReference struct {
 	FieldPath string `json:"fieldPath,omitempty"`
 }
 
+type SerializedReference struct {
+	TypeMeta `json:",inline"`
```
Comment (lavalamp) [inline pkg/api/types.go]: ObjectMeta

File: pkg/api/v1beta3/types.go [hunk 2]
```diff
@@ -1515,6 +1515,11 @@ type ObjectReference struct {
 	FieldPath string `json:"fieldPath,omitempty" description:"if referring to a piece of an object instead of an entire object, this string should contain a valid JSON/Go field access statement, such as desiredState.manifest.containers[2]"`
 }
 
+type SerializedReference struct {
+	TypeMeta `json:",inline"`
```
Comment (lavalamp) [inline pkg/api/v1beta3/types.go]: ObjectMeta & TypeMeta

**Comment by lavalamp** [comment]
This looks mostly correct, thanks-- just need to follow objectmeta/typemeta conventions-- they differ a little between versions.

**Comment by simon3z** [comment]
@lavalamp I am just using Kind/APIVersion, I think TypeMeta is the correct one for all the versions. Isn't it?

**Comment by lavalamp** [comment]
Ah-- actually I think you might be right. We don't need object meta, since we don't store this as a top level object.

**Comment by smarterclayton** [comment]
Brian was proposing the dependencies object for this use case, but it doesn't exist yet.  That would be a list of object references (or object references with additional data).

**Comment by lavalamp** [comment]
LGTM, needs a rebase.

**Comment by simon3z** [comment]
@lavalamp I have no idea why Shippable failed. Anyway I expect travis to pass.

**Comment by simon3z** [comment]
@lavalamp ping, can this be merged?

**Comment by lavalamp** [comment]
Sorry for delay. We've added the pkg/api/v1 api version, can you add this in there, too?

**Comment by simon3z** [comment]
@lavalamp can you check if you can merge this today. Thanks.

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
Will merge on green

**Pull request merged**